.github/workflows/main.yml

```diff
@@ -37,6 +37,7 @@ jobs:
       - name: unit test
         run: make test
       - name: Code coverage
-        uses: codecov/codecov-action@v3
+        uses: codecov/codecov-action@v4
         with:
-         files: build/coverage.txt
+          files: build/coverage.txt
+          token: ${{ secrets.CODECOV_TOKEN }}
```

Comment:
@chenyulin0719 Please explain why we need this change? thanks!

Reply:
Oh, I just put the comment above.
> Hi @chia7712,
In order to support build in our personal repo.
I added one more commit to set secrets.CODECOV_TOKEN.
(Everyone can follow the [doc](https://docs.codecov.com/docs/adding-the-codecov-token) to set codecov in their forked repo.)

Reply:
> In order to support build in our personal repo.
I added one more commit to set secrets.CODECOV_TOKEN.

got it. Could you add comments to the source code?

Reply:
Sure.